backend/app/adapter/google/account_integration_test.go

```diff
@@ -0,0 +1,91 @@
+package google
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/short-d/app/mdtest"
+	"github.com/short-d/short/app/entity"
+)
+
+func TestAccount_GetSingleSignOnUser(t *testing.T) {
+	testCases := []struct {
+		name            string
+		httpResponse    *http.Response
+		httpErr         error
+		expectHasErr    bool
+		expectedSSOUser entity.SSOUser
+	}{
+		{
+			name:         "invalid access token",
+			httpResponse: nil,
+			httpErr:      errors.New("invalid access token"),
+			expectHasErr: true,
+		},
+		{
+			name: "user has id, email, and name",
+			httpResponse: &http.Response{
+				StatusCode: http.StatusOK,
+				Body: ioutil.NopCloser(bytes.NewReader([]byte(`
+{
+      "sub": "bcBi3AMeOV3Zg3AlOPyn",
+      "name": "Google User",
+      "email": "googleUser@gmail.com"
+}
+`,
+				)))},
+			expectHasErr: false,
+			expectedSSOUser: entity.SSOUser{
+				ID:    "bcBi3AMeOV3Zg3AlOPyn",
+				Name:  "Google User",
+				Email: "googleUser@gmail.com",
+			},
+		},
+		{
+			name: "user doesn't have email",
+			httpResponse: &http.Response{
+				StatusCode: http.StatusOK,
+				Body: ioutil.NopCloser(bytes.NewReader([]byte(`
+{
+      "sub": "bcBi3AMeOV3Zg3AlOPyn",
+      "name": "Google User",
+      "email": ""
+}
+`,
+				)))},
+			expectHasErr: false,
+			expectedSSOUser: entity.SSOUser{
+				ID:    "bcBi3AMeOV3Zg3AlOPyn",
+				Name:  "Google User",
+				Email: "",
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			httpRequest := mdtest.NewHTTPRequestFake(
+				func(req *http.Request) (response *http.Response, e error) {
+					mdtest.Equal(t, "https://openidconnect.googleapis.com/v1/userinfo", req.URL.String())
```

Comment:
If you checked request URL here, maybe it will be a good idea to check that `access token` has been added to headers?

Reply:
We should check access token in header because it's part of Google's API spec

Reply:
addressed comment